### PR TITLE
Add the star vampire, an invisible blood-drinking monster

### PR DESCRIPTION
**Pull request by Standing-Storm**
#### Summary
Content "Add the star vampire, an invisible blood-drinking monster"

#### Purpose of change

I've been wanting to add a blood-drinking monster ever since I learned how effect vitamin changes work, and this is a good chance.

#### Describe the solution

Add the star vampire, an invisible blood-drinking monster.  I'm hoping that monsters can benefit from INVISIBLE flag effects--if not I'll have to give them NIGHT_INVISIBILITY and CAMOUFLAGE.

Update: Monsters cannot benefit from the INVISIBLE flag and my initial attempt to allow them to didn't work.  
Update update: I figured it out.  Prepare to be jump-scared.

The star vampire charges in, grabs the target, and once a grab is established, starts drinking blood (represented by an effect that removes your hidden blood vitamin).  Once the star vampire has drunk blood, an effect is placed on it that removes its invisibility and prevents it from re-applying it.  Otherwise it's similar to the flying polyp but slower and without hit and run. While invisible, it can still be detected by the high-pitched tittering it emits, seemingly uncontrollably.  If you wear thick enough armor with good coverage, you can prevent its hideous tentacles from being able to drink your blood.

The star vampire normally has no regeneration, but while drinking blood and for some time after it has regeneration 5.

I wanted to make it resistant to bullets but if the flying polyp has no bullet resistance, this thing definitely shouldn't either.

#### Testing

Went to the nether monster corpse and waited a while near the portals.  A star vampire came out.

Debug-mutated gastropod, the star vampire did not attempt to drink my blood.  Debug-mutated acid blood, the star vampire took damage when biting me. 

Combat testing:
Evacuee: No contest, all that was left was a withered, exsanguinated corpse.
Scavenger with fire axe: Death, did barely any damage.  Same with shillelagh. National guard bayonet was able to kill it with minimal blood drunk or damage. Lesson--if a horrific tentacled monster is attacking you, do not use two-handed weapons.
Survivor: Easily killed it. Thanks to late-game armor, I was even able to run back to the evac shelter and grab a rapier after it disarmed my nodachi, and the star vampire was not able to penetrate my armor. 

#### Additional context

I originally wanted to give it a different name, but since it's part of the lore that the XEDRA scientists were nerds who named everything after Lovecraft, I gave it the Lovecraft name.

I made spawns relatively low because "invisible monster that can appear right next to you while you're engaging in combat already and has two grabs" is a serious threat.

### CODE REVIEW

**Comment by fairyarmadillo** [comment]
This looks awesome. You may want to give it the HARDTOSHOOT flag, which will make it a bit harder to hit. Seems like a good candidate for the flag.

File: data/json/monsters/nether.json [hunk 1]
```diff
@@ -533,6 +533,82 @@
...
+        "effects": [ { "id": "star_vampire_blood_drink", "duration": 1, "chance": 100 } ],
+        "self_effects_ondmg": [ { "id": "star_vampire_blood_drank", "duration": 10000 } ],
+        "cooldown": 1,
+        "move_cost": 100,
+        "hit_dmg_u": "%1$s fastens onto your %2$s and greedily drinks your blood!",
+        "hit_dmg_npc": "%1$s fastens onto <npcname>'s %2$s and greedily drinks their blood!",
+        "miss_msg_u": "%1$s tries to fasten onto you, but you dodge!",
+        "miss_msg_npc": "%1$s tries to fasten onto you <npcname>, but they dodge!",
+        "no_dmg_msg_u": "%1$s tries to fasten onto your %2$s without penetrating your armor.",
+        "no_dmg_msg_npc": "%1$s  tries to fasten onto <npcname>'s %2$s without penetrating their armor."
```
Comment (NetSysFire) [inline data/json/monsters/nether.json:580]:
```suggestion
        "no_dmg_msg_npc": "%1$s tries to fasten onto <npcname>'s %2$s without penetrating their armor."
```

**Comment by Standing-Storm** [comment]
Alright, blood-drinking is in and works...most of the time. It always decides to drink player and NPC blood, and it's appropriately nasty, but the special attack I put in to simulate drinking feral blood just sometimes...doesn't fire and it turns into a slapfight. And then sometimes it does and the star vampire drains the feral dry. No idea why.

Unfortunately, monsters can't benefit from the INVISIBLE flag and my attempt to account for the Invisibility effect in creature.cpp isn't working, so that needs more thought.

**Comment by fairyarmadillo** [comment]
> Alright, blood-drinking is in and works...most of the time. It always decides to drink player and NPC blood, and it's appropriately nasty, but the special attack I put in to simulate drinking feral blood just sometimes...doesn't fire and it turns into a slapfight. And then sometimes it does and the star vampire drains the feral dry. No idea why.

Would the star vampire be appropriately injured by drinking from characters with acidic blood? What about arthropod blood? Mutant players can get either of these.

**Comment by Inglonias** [comment]
I'm not a fan on the grounds that this seems like a bullshit death waiting to happen in the early game. I thought we were moving away from that?

**Comment by Maleclypse** [comment]
> I'm not a fan on the grounds that this seems like a bullshit death waiting to happen in the early game. I thought we were moving away from that?

I think in the future you should read an entire PR a second time before you throw the term "bullshit death waiting to happen in the early game" around.  This only appears in labs, inside of kaiju corpses and after the player has incurred significant teleglow.  None of those things are early bullshit deaths, they are all results of the player fucking around and finding out.  I appreciate your desire to protect early game players but it appears misplaced in this PR.

**Comment by Inglonias** [comment]
My apologies. That fact was not in the PR description on GitHub and was missed.

Carry on.

**Pull request merged**